docs/source/developer_zone/changes-2.0.rst

```diff
@@ -40,6 +40,9 @@ Breaking changes include:
 - A new config ``order_solver_request`` (default true) can be used to order the dependencies passed
   to the solver, getting order independent solutions.
 - Support for complex match specs such as ``pkg[md5=0000000000000]`` and ``pkg[build='^\d*$']``.
+- Dropped support for leading and internal globs in version strings (via
+  redesigned ``VersionSpec``, which no longer handles version strings as a
+  regex). Only trailing globs continue to be supported.
```

Comment:
I'm not sure we should advertise that leading and internal globs are definitely dropped as they may be supported later on...
An opinion @JohanMabille?

Reply:
Indeed, this is more a regression than a voluntary drop.

Reply:
I appreciate that the plan is to fix this regression (and I would love that!), but [mamba 2](https://github.com/mamba-org/mamba/releases/tag/2024.09.25) was released about 4 months ago. Thus I think we need to better document this regression since this change in behavior was unexpected (eg https://github.com/mamba-org/mamba/issues/3601). Some ideas:

* What if we created a new Issue (something like "bug: fix leading and internal globs in version strings") and then pin that to the top of the Issues page?
* I could update the language in this PR to clarify that this breaking change is a temporary regression. Would that work? While it is only temporary, I feel like it deserves to be documented somewhere in the list of breaking changes.

Reply:
@jdblischak Yes the comments were only meant to say that we shouldn't document that this is dropped, only that it's a regression that we are aiming to fix. But it's definitely worth documenting!

Reply:
@Hind-M @JohanMabille I have updated the language to reflect the fact that this breaking change was unintentional and only temporary. I'm happy to update it further based on your preferred phrasing.